src/internal/NeoEsp8266UartMethod.cpp

```diff
@@ -0,0 +1,209 @@
+    // when it will finish.
+    _startTime = micros();
+
+    // Then keep filling the FIFO until done
+    const uint8_t* ptr = _pixels;
+    const uint8_t* end = ptr + _sizePixels;
+    while (ptr != end)
+    {
+        ptr = FillUartFifo(ptr, end);
+    }
+}
+
+const uint8_t* ICACHE_RAM_ATTR NeoEsp8266Uart::FillUartFifo(const uint8_t* pixels, const uint8_t* end)
+{
+    // Remember: UARTs send less significant bit (LSB) first so
+    //      pushing ABCDEF byte will generate a 0FEDCBA1 signal,
+    //      including a LOW(0) start & a HIGH(1) stop bits.
+    // Also, we have configured UART to invert logic levels, so:
+    const uint8_t _uartData[4] = {
+        0b110111, // On wire: 1 000 100 0 [Neopixel reads 00]
+        0b000111, // On wire: 1 000 111 0 [Neopixel reads 01]
+        0b110100, // On wire: 1 110 100 0 [Neopixel reads 10]
+        0b000100, // On wire: 1 110 111 0 [NeoPixel reads 11]
+    };
+    uint8_t avail = (UART_TX_FIFO_SIZE - getUartTxFifoLength()) / 4;
+    if (end - pixels > avail)
+    {
+        end = pixels + avail;
+    }
+    while (pixels < end)
+    {
+        uint8_t subpix = *pixels++;
+        U1F = _uartData[(subpix >> 6) & 3];
+        U1F = _uartData[(subpix >> 4) & 3];
+        U1F = _uartData[(subpix >> 2) & 3];
+        U1F = _uartData[subpix & 3];
+    }
+    return pixels;
+}
+
+NeoEsp8266AsyncUart::NeoEsp8266AsyncUart(uint8_t pin, uint16_t pixelCount, size_t elementSize)
+    : NeoEsp8266Uart(pin, pixelCount, elementSize)
+{
+    _asyncPixels = (uint8_t*)malloc(this->_sizePixels);
+}
+
+NeoEsp8266AsyncUart::~NeoEsp8266AsyncUart()
+{
+    // Remember: the UART interrupt can be sending data from _asyncPixels in the background
+    while (esp8266_uart1_async_buf != esp8266_uart1_async_buf_end)
+    {
+        yield();
+    }
+    free(_asyncPixels);
+}
+
+void ICACHE_RAM_ATTR NeoEsp8266AsyncUart::InitializeUart(uint32_t uartBaud)
+{
+    NeoEsp8266Uart::InitializeUart(uartBaud);
+
+    // Disable all interrupts
+    ETS_UART_INTR_DISABLE();
+
+    // Clear the RX & TX FIFOS
+    SET_PERI_REG_MASK(UART_CONF0(UART1), UART_RXFIFO_RST | UART_TXFIFO_RST);
+    CLEAR_PERI_REG_MASK(UART_CONF0(UART1), UART_RXFIFO_RST | UART_TXFIFO_RST);
+
+    // Set the interrupt handler
+    ETS_UART_INTR_ATTACH(IntrHandler, NULL);
+
+    // Set tx fifo trigger. 80 bytes gives us 200 microsecs to refill the FIFO
+    WRITE_PERI_REG(UART_CONF1(UART1), 80 << UART_TXFIFO_EMPTY_THRHD_S);
+
+    // Disable RX & TX interrupts. It is enabled by uart.c in the SDK
+    CLEAR_PERI_REG_MASK(UART_INT_ENA(UART1), UART_RXFIFO_FULL_INT_ENA | UART_TXFIFO_EMPTY_INT_ENA);
+
+    // Clear all pending interrupts in UART1
+    WRITE_PERI_REG(UART_INT_CLR(UART1), 0xffff);
+
+    // Reenable interrupts
+    ETS_UART_INTR_ENABLE();
+}
+
+void NeoEsp8266AsyncUart::UpdateUart()
+{
+    // Instruct ESP8266 hardware uart1 to send the pixels asynchronously
+    esp8266_uart1_async_buf = this->_pixels;
```

Comment:
this is problematic, the pixels buffer is exposed so that the consumer can manipulate it directly (it has a fixed format) for any advanced pixel manipulation.  But you are swapping the pointer on them.